python/ray/data/preprocessor.py

```diff
@@ -63,10 +63,21 @@ class FitStatus(str, Enum):
     # Preprocessors that do not need to be fitted must override this.
     _is_fittable = True
 
+    def _check_has_fitted_state(self):
+        """Checks if the Preprocessor has fitted state.
+
+        This is also used as an indiciation if the Preprocessor has been fit.
+        """
+
```

Comment:
If you could add a note here explaining why we need this (and maybe link the relevant PR), I think that'd be helpful. I'm not sure if I'd understand this method without the context.